paasta_tools/cli/cmds/local_run.py

```diff
@@ -727,10 +727,26 @@ def assume_aws_role(
         )
     elif use_okta_role:
         print(f"Calling aws-okta using account {aws_account}")
+    elif "AWS_ROLE_ARN" in os.environ and "AWS_WEB_IDENTITY_TOKEN_FILE" in os.environ:
+        # Get a session using the current pod identity
+        print(
+            f"Found Pod Identity token in env. Assuming into role {os.environ['AWS_ROLE_ARN']}."
+        )
+        boto_session = boto3.Session()
+        credentials = boto_session.get_credentials()
+        creds_dict: AWSSessionCreds = {
+            "AWS_ACCESS_KEY_ID": credentials.access_key,
+            "AWS_SECRET_ACCESS_KEY": credentials.secret_key,
+            "AWS_SESSION_TOKEN": credentials.token,
+            "AWS_SECURITY_TOKEN": credentials.token,
```

Comment:
this is for boto2 compatibility? I thought web identity didn't work with boto2?

Reply:
Yeah, it's probably time to get rid of this. Especially because, as you point out, boto2 doesn't support pod identity, so you wouldn't need it to work while emulating pod identity...